directive/authorized.go

```diff
@@ -55,102 +56,114 @@ func extractNestedKeyFromArgs(args map[string]any, paramName string) (string, er
+			return check, sentry.SentryError(err)
 		}
 
-		token, err := openmfpcontext.GetAuthHeaderFromContext(ctx)
-		hasToken := err == nil
+		return check, err
+	}
+}
 
-		if hasToken {
-			ctx = metadata.AppendToOutgoingContext(ctx, "authorization", token)
-		}
+func executeTheAuthCheck(ctx context.Context, openfgaClient openfgav1.OpenFGAServiceClient, log *logger.Logger, entityParamName string, entityTypeParamName *string, entityType *string, relation string, next graphql.Resolver) (interface{}, error) {
+	if openfgaClient == nil {
+		return nil, errors.New("OpenFGAServiceClient is nil. Cannot process request")
+	}
 
-		fctx := graphql.GetFieldContext(ctx)
+	ctx, err := setTenantToContextForTechnicalUsers(ctx, log)
+	if err != nil {
+		return nil, err
+	}
 
-		entityID, err := extractNestedKeyFromArgs(fctx.Args, entityParamName)
-		if err != nil {
-			return nil, err
-		}
+	token, err := openmfpcontext.GetAuthHeaderFromContext(ctx)
+	hasToken := err == nil
 
-		tenantID, err := openmfpcontext.GetTenantFromContext(ctx)
-		if err != nil {
-			return nil, err
-		}
+	if hasToken {
+		ctx = metadata.AppendToOutgoingContext(ctx, "authorization", token)
+	}
 
-		evaluatedEntityType := ""
-		if entityTypeParamName != nil {
-			evaluatedEntityType, err = extractNestedKeyFromArgs(fctx.Args, *entityTypeParamName)
-			if err != nil {
-				return nil, err
-			}
-		} else if entityType != nil {
-			evaluatedEntityType = *entityType
-		}
+	fctx := graphql.GetFieldContext(ctx)
 
-		if evaluatedEntityType == "" {
-			return nil, fmt.Errorf("make sure to either provide entityType or entityTypeParamName")
-		}
+	entityID, err := extractNestedKeyFromArgs(fctx.Args, entityParamName)
+	if err != nil {
+		return nil, err
+	}
 
-		storeID, err := helpers.GetStoreIDForTenant(ctx, openfgaClient, tenantID)
-		if err != nil {
-			return nil, err
-		}
-		modelID, err := helpers.GetModelIDForTenant(ctx, openfgaClient, tenantID)
+	tenantID, err := openmfpcontext.GetTenantFromContext(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	evaluatedEntityType := ""
+	if entityTypeParamName != nil {
+		evaluatedEntityType, err = extractNestedKeyFromArgs(fctx.Args, *entityTypeParamName)
 		if err != nil {
 			return nil, err
 		}
+	} else if entityType != nil {
+		evaluatedEntityType = *entityType
+	}
 
-		var userID string
-		if hasToken {
-			user, err := openmfpcontext.GetWebTokenFromContext(ctx)
-			if err != nil {
-				return nil, err
-			}
-			userID = user.Subject
-		} else {
-			spiffe, err := openmfpcontext.GetSpiffeFromContext(ctx)
-			if err != nil {
-				return nil, fmt.Errorf("authorized was invoked without a user token or a spiffe header")
-			}
-			userID = strings.TrimPrefix(spiffe, "spiffe://")
-			log.Trace().Str("user", userID).Msg("using spiffe user in authorized directive")
-		}
+	if evaluatedEntityType == "" {
+		return nil, fmt.Errorf("make sure to either provide entityType or entityTypeParamName")
```

Comment:
Consider using an exported error var (or at least not mix `fmt.Errorf` and `errors.New` from above)